e2e/test/iothub/method/MethodE2ETests.cs

```diff
@@ -477,12 +477,12 @@ public static async Task<Task> SetDeviceReceiveMethodDefaultHandlerAsync(DeviceC
             return methodCallReceived.Task;
         }
 
-        private static Task<Task> SetDeviceReceiveMethodObsoleteHandler(DeviceClient deviceClient, string methodName, MsTestLogger logger)
+        private static async Task<Task> SetDeviceReceiveMethodObsoleteHandler(DeviceClient deviceClient, string methodName, MsTestLogger logger)
```

Comment:
nit: we could rename the test method here to remove the "obsolete" keyword and add the "async" suffix.

Reply:
On that note, we should be having an existing test method for testing this device client method, so you might be able to simply delete this one.